runtime/src/main/java/org/corfudb/runtime/object/transactions/SnapshotTransactionalContext.java

```diff
@@ -32,8 +32,8 @@
     @Getter
     private Set<ICorfuSMRProxyInternal> modifiedProxies = ImmutableSet.of();
 
-    public SnapshotTransactionalContext(TransactionBuilder builder) {
-        super(builder);
+    public SnapshotTransactionalContext(Transaction transaction) {
+        super(transaction);
```

Comment:
Please pass `CorfuRuntime` as a parameter, don't use transaction.getRuntime

Reply:
Transactions are tied to a particular runtime.

Reply:
Just renaming, not changing the logic of tryCommitAllProxies